src/Generator/Schema/Field.cs

```diff
@@ -31,19 +31,38 @@ public string ClrType
 			get
 			{
 				// Special cases.
-				if (Name == "args") return "string[]";
-				if (Name == "data.strings") return "string[]";
-				if (Name == "parent.args") return "string[]";
-				if (Name.Contains("certificate_chain")) return "string[]";
-				if (Name.Contains("supported_ciphers")) return "string[]";
-				if (Schema.Name == "vulnerability" && Name == "category") return "string[]";
-				if (Schema.Name == "file" && Name == "attributes") return "string[]";
-				if (Schema.Name == "dns" && Name == "header_flags") return "string[]";
-				if (Schema.Name == "dns" && Name == "resolved_ip") return "string[]";
-				if (Schema.Name == "user" && FlatName == "user.id") return "string[]";
-				if (Schema.Name == "base" && Name == "tags") return "string[]";
-				if (Schema.Name == "base" && Name == "labels") return "IDictionary<string, object>";
-				if (Schema.Name == "base" && Name == "_metadata") return "IDictionary<string, object>";
+				if (FlatName == "container.image.tag") return "string[]";
```

Comment:
switch expression :)

Reply:
No doubt, it is ugly, I agree - when https://github.com/elastic/ecs/pull/727 lands this whole section will be deleted...